internal/abi/fastly/httpcache_guest.go

```diff
@@ -0,0 +1,1107 @@
+//go:build ((tinygo.wasm && wasi) || wasip1) && !nofastlyhostcalls
+
+package fastly
+
+import (
+	"github.com/fastly/compute-sdk-go/internal/abi/prim"
+)
+
+const TRACE = true
+
+type HTTPCacheLookupOptions struct {
+	mask httpCacheLookupOptionsMask
+	opts httpCacheLookupOptions
+}
+
+func (o *HTTPCacheLookupOptions) OverrideKey(key string) {
+	k := []byte(key)
+	o.mask |= httpCacheLookupOptionsFlagOverrideKey
+	buf := prim.NewReadBufferFromBytes(k)
+	o.opts.overrideKeyPtr = prim.ToPointer(buf.Char8Pointer())
+	o.opts.overrideKeyLen = buf.Len()
```

Comment:
```suggestion
	buf := prim.NewReadBufferFromBytes(k)
	o.opts.overrideKeyPtr = prim.ToPointer(buf.Char8Pointer())
	o.opts.overrideKeyLen = buf.Len()
	o.mask |= httpCacheLookupOptionsFlagOverrideKey
```
to be more consistent with similar functions that follow.

Reply:
c39aae2